SOC2 CI-100 - test environment for MIs described in handbook (#3744)
sourcegraph/handbook: content/departments/product-engineering/engineering/cloud/devops/managed/index.md

We do not offer Kubernetes managed instances today as this introduces some complexity for us in terms of ongoing maintenance and overhead, we may revisit this decision in the future.

> ### Environments

Author: @michaellzc I moved your docs regarding testing environment from operations to main MI page (agreed with Raf that it's very important so should live on first page) and added some small extension for v1.0 & v1.1 - hopefully I moved all correct.

Reviewer: yep! this looks good to me

<span class="badge badge-note">SOC2/CI-100</span>

### Internal instances

For each type of Managed Instances (v1.0 and v.1.1), Souregraph maintains separate test environments:

- for v1.0 - [dev instance](https://devmanaged.sourcegraph.com/)
- for v1.1 - [rctest instance](https://rctest.sourcegraph.com/)

Internal instances are created for various testing purposes:

> - testing changes prior to the montly upgrade on customer instances, e.g <https://devmanaged.sourcegraph.com>
> - testing siginificant opertional changes prior applying to customer instances

Reviewer 2: few typos
```suggestion
- testing changes prior to the monthly upgrade on customer instances, e.g <https://devmanaged.sourcegraph.com>
- testing significant operational changes prior to applying to customer instances
```

- short-lived instances for product teams to test important product changes. Notes: any teammate may request a managed instance through our [request process](./index.md#managed-instance-requests)

### Customer instances

> All customer instances are considered production enviornment and all changes applied to these customers should be well-tested in internal environment.

Reviewer 2: ```suggestion
All customer instances are considered part of the production environment and all changes applied to these customers should be well-tested in the dev/preprod environment.
```
